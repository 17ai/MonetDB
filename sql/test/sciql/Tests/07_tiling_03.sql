create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f tinyint);
insert into a values (0,0,2), (0,1,5), (0,2,2), (1,0,5), (1,1,1.0), (1,2,5), (2,0,2), (2,1,5), (2,2,2); 
select x,y, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) as f_avg from a group by a[x:x+2][y:y+2];
drop array a;

create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f smallint);
insert into a values (0,0,2), (0,1,5), (0,2,2), (1,0,5), (1,1,1.0), (1,2,5), (2,0,2), (2,1,5), (2,2,2); 
select x,y, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) as f_avg from a group by a[x:x+2][y:y+2];
drop array a;

create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f integer);
insert into a values (0,0,2), (0,1,5), (0,2,2), (1,0,5), (1,1,1.0), (1,2,5), (2,0,2), (2,1,5), (2,2,2); 
select x,y, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) as f_avg from a group by a[x:x+2][y:y+2];
drop array a;

create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f bigint);
insert into a values (0,0,2), (0,1,5), (0,2,2), (1,0,5), (1,1,1.0), (1,2,5), (2,0,2), (2,1,5), (2,2,2); 
select x,y, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) as f_avg from a group by a[x:x+2][y:y+2];
drop array a;

create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f real);
insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
select x,y, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) as f_avg from a group by a[x:x+2][y:y+2];
drop array a;

create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f double);
insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
select x,y, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) from a group by a[x:x+2][y:y+2];
select x,y, f, avg(f * f), min(f * f), max(f * f) as f_avg from a group by a[x:x+2][y:y+2];
drop array a;

