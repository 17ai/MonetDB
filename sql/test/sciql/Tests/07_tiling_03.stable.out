stdout of test '07_tiling_03` in directory 'sql/test/sciql` itself:


# 23:28:26 >  
# 23:28:26 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/export/scratch0/zhang/monet-install/sciql/debug/var/MonetDB" "--set" "mapi_open=true" "--set" "mapi_port=33271" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_test_sciql" "--set" "mal_listing=0"
# 23:28:26 >  

# MonetDB 5 server v11.8.0
# This is an unreleased version
# Serving database 'mTests_test_sciql', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 7.749 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2012 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://jarl.ins.cwi.nl:33271/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_history.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_compress.sql
# loading sql script: 18_dictionary.sql
# loading sql script: 19_cluster.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 24_zorder.sql
# loading sql script: 25_debug.sql
# loading sql script: 29_array.sql
# loading sql script: 40_geom.sql
# loading sql script: 80_udf.sql
# loading sql script: 99_system.sql

# 23:28:26 >  
# 23:28:26 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=jarl" "--port=33271"
# 23:28:26 >  

#create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f float);
#insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
[ 9	]
#select x,y,    avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	smallint,	smallint,	bigint # type
% 1,	1,	24,	1,	2,	1 # length
[ 0,	0,	13.75,	1,	25,	4	]
[ 0,	1,	13.75,	1,	25,	4	]
[ 0,	2,	14.5,	4,	25,	2	]
[ 1,	0,	13.75,	1,	25,	4	]
[ 1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	14.5,	4,	25,	2	]
[ 2,	0,	14.5,	4,	25,	2	]
[ 2,	1,	14.5,	4,	25,	2	]
[ 2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	L1,	L2,	L3,	L4 # name
% int,	int,	tinyint,	double,	smallint,	smallint,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f) as ff_avg, min(f * f) as ff_min, max(f * f) as ff_max, count(f * f) as f_cnt from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	ff_avg,	ff_min,	ff_max,	f_cnt # name
% int,	int,	tinyint,	double,	smallint,	smallint,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#drop array a;

#create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f float);
#insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
[ 9	]
#select x,y,    avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	int,	int,	bigint # type
% 1,	1,	24,	1,	2,	1 # length
[ 0,	0,	13.75,	1,	25,	4	]
[ 0,	1,	13.75,	1,	25,	4	]
[ 0,	2,	14.5,	4,	25,	2	]
[ 1,	0,	13.75,	1,	25,	4	]
[ 1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	14.5,	4,	25,	2	]
[ 2,	0,	14.5,	4,	25,	2	]
[ 2,	1,	14.5,	4,	25,	2	]
[ 2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	L1,	L2,	L3,	L4 # name
% int,	int,	smallint,	double,	int,	int,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f) as ff_avg, min(f * f) as ff_min, max(f * f) as ff_max, count(f * f) as f_cnt from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	ff_avg,	ff_min,	ff_max,	f_cnt # name
% int,	int,	smallint,	double,	int,	int,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#drop array a;

#create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f float);
#insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
[ 9	]
#select x,y,    avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	bigint,	bigint,	bigint # type
% 1,	1,	24,	1,	2,	1 # length
[ 0,	0,	13.75,	1,	25,	4	]
[ 0,	1,	13.75,	1,	25,	4	]
[ 0,	2,	14.5,	4,	25,	2	]
[ 1,	0,	13.75,	1,	25,	4	]
[ 1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	14.5,	4,	25,	2	]
[ 2,	0,	14.5,	4,	25,	2	]
[ 2,	1,	14.5,	4,	25,	2	]
[ 2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	L1,	L2,	L3,	L4 # name
% int,	int,	int,	double,	bigint,	bigint,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f) as ff_avg, min(f * f) as ff_min, max(f * f) as ff_max, count(f * f) as f_cnt from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	ff_avg,	ff_min,	ff_max,	f_cnt # name
% int,	int,	int,	double,	bigint,	bigint,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#drop array a;

#create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f float);
#insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
[ 9	]
#select x,y,    avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	bigint,	bigint,	bigint # type
% 1,	1,	24,	1,	2,	1 # length
[ 0,	0,	13.75,	1,	25,	4	]
[ 0,	1,	13.75,	1,	25,	4	]
[ 0,	2,	14.5,	4,	25,	2	]
[ 1,	0,	13.75,	1,	25,	4	]
[ 1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	14.5,	4,	25,	2	]
[ 2,	0,	14.5,	4,	25,	2	]
[ 2,	1,	14.5,	4,	25,	2	]
[ 2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	L1,	L2,	L3,	L4 # name
% int,	int,	bigint,	double,	bigint,	bigint,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#select x,y, f, avg(f * f) as ff_avg, min(f * f) as ff_min, max(f * f) as ff_max, count(f * f) as f_cnt from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	ff_avg,	ff_min,	ff_max,	f_cnt # name
% int,	int,	bigint,	double,	bigint,	bigint,	bigint # type
% 1,	1,	1,	24,	1,	2,	1 # length
[ 0,	0,	2,	13.75,	1,	25,	4	]
[ 0,	1,	5,	13.75,	1,	25,	4	]
[ 0,	2,	2,	14.5,	4,	25,	2	]
[ 1,	0,	5,	13.75,	1,	25,	4	]
[ 1,	1,	1,	13.75,	1,	25,	4	]
[ 1,	2,	5,	14.5,	4,	25,	2	]
[ 2,	0,	2,	14.5,	4,	25,	2	]
[ 2,	1,	5,	14.5,	4,	25,	2	]
[ 2,	2,	2,	4,	4,	4,	1	]
#drop array a;

#create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f float);
#insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
[ 9	]
#select x,y,    avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	real,	real,	bigint # type
% 1,	1,	24,	15,	15,	1 # length
[ 0,	0,	0.3850000007,	0.0400000028,	1,	4	]
[ 0,	1,	0.3850000007,	0.0400000028,	1,	4	]
[ 0,	2,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 1,	0,	0.3850000007,	0.0400000028,	1,	4	]
[ 1,	1,	0.3850000007,	0.0400000028,	1,	4	]
[ 1,	2,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	0,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	1,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	2,	0.04000000283,	0.0400000028,	0.0400000028,	1	]
#select x,y, f, avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	L1,	L2,	L3,	L4 # name
% int,	int,	real,	double,	real,	real,	bigint # type
% 1,	1,	15,	24,	15,	15,	1 # length
[ 0,	0,	0.200000003,	0.3850000007,	0.0400000028,	1,	4	]
[ 0,	1,	0.5,	0.3850000007,	0.0400000028,	1,	4	]
[ 0,	2,	0.200000003,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 1,	0,	0.5,	0.3850000007,	0.0400000028,	1,	4	]
[ 1,	1,	1,	0.3850000007,	0.0400000028,	1,	4	]
[ 1,	2,	0.5,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	0,	0.200000003,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	1,	0.5,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	2,	0.200000003,	0.04000000283,	0.0400000028,	0.0400000028,	1	]
#select x,y, f, avg(f * f) as ff_avg, min(f * f) as ff_min, max(f * f) as ff_max, count(f * f) as f_cnt from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	ff_avg,	ff_min,	ff_max,	f_cnt # name
% int,	int,	real,	double,	real,	real,	bigint # type
% 1,	1,	15,	24,	15,	15,	1 # length
[ 0,	0,	0.200000003,	0.3850000007,	0.0400000028,	1,	4	]
[ 0,	1,	0.5,	0.3850000007,	0.0400000028,	1,	4	]
[ 0,	2,	0.200000003,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 1,	0,	0.5,	0.3850000007,	0.0400000028,	1,	4	]
[ 1,	1,	1,	0.3850000007,	0.0400000028,	1,	4	]
[ 1,	2,	0.5,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	0,	0.200000003,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	1,	0.5,	0.1450000014,	0.0400000028,	0.25,	2	]
[ 2,	2,	0.200000003,	0.04000000283,	0.0400000028,	0.0400000028,	1	]
#drop array a;

#create array a ( x integer dimension[0:1:3], y integer dimension[0:1:3], f float);
#insert into a values (0,0,0.2), (0,1,0.5), (0,2,0.2), (1,0,0.5), (1,1,1.0), (1,2,0.5), (2,0,0.2), (2,1,0.5), (2,2,0.2); 
[ 9	]
#select x,y,    avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	double,	double,	bigint # type
% 1,	1,	24,	24,	24,	1 # length
[ 0,	0,	0.385,	0.04,	1,	4	]
[ 0,	1,	0.385,	0.04,	1,	4	]
[ 0,	2,	0.145,	0.04,	0.25,	2	]
[ 1,	0,	0.385,	0.04,	1,	4	]
[ 1,	1,	0.385,	0.04,	1,	4	]
[ 1,	2,	0.145,	0.04,	0.25,	2	]
[ 2,	0,	0.145,	0.04,	0.25,	2	]
[ 2,	1,	0.145,	0.04,	0.25,	2	]
[ 2,	2,	0.04,	0.04,	0.04,	1	]
#select x,y, f, avg(f * f), min(f * f), max(f * f), count(f * f) from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	L1,	L2,	L3,	L4 # name
% int,	int,	double,	double,	double,	double,	bigint # type
% 1,	1,	24,	24,	24,	24,	1 # length
[ 0,	0,	0.2,	0.385,	0.04,	1,	4	]
[ 0,	1,	0.5,	0.385,	0.04,	1,	4	]
[ 0,	2,	0.2,	0.145,	0.04,	0.25,	2	]
[ 1,	0,	0.5,	0.385,	0.04,	1,	4	]
[ 1,	1,	1,	0.385,	0.04,	1,	4	]
[ 1,	2,	0.5,	0.145,	0.04,	0.25,	2	]
[ 2,	0,	0.2,	0.145,	0.04,	0.25,	2	]
[ 2,	1,	0.5,	0.145,	0.04,	0.25,	2	]
[ 2,	2,	0.2,	0.04,	0.04,	0.04,	1	]
#select x,y, f, avg(f * f) as ff_avg, min(f * f) as ff_min, max(f * f) as ff_max, count(f * f) as ff_cnt from a group by a[x:x+2][y:y+2];
% sys.a,	sys.a,	sys.a,	sys.,	sys.,	sys.,	sys. # table_name
% x,	y,	f,	ff_avg,	ff_min,	ff_max,	ff_cnt # name
% int,	int,	double,	double,	double,	double,	bigint # type
% 1,	1,	24,	24,	24,	24,	1 # length
[ 0,	0,	0.2,	0.385,	0.04,	1,	4	]
[ 0,	1,	0.5,	0.385,	0.04,	1,	4	]
[ 0,	2,	0.2,	0.145,	0.04,	0.25,	2	]
[ 1,	0,	0.5,	0.385,	0.04,	1,	4	]
[ 1,	1,	1,	0.385,	0.04,	1,	4	]
[ 1,	2,	0.5,	0.145,	0.04,	0.25,	2	]
[ 2,	0,	0.2,	0.145,	0.04,	0.25,	2	]
[ 2,	1,	0.5,	0.145,	0.04,	0.25,	2	]
[ 2,	2,	0.2,	0.04,	0.04,	0.04,	1	]
#drop array a;

# 23:28:26 >  
# 23:28:26 >  "Done."
# 23:28:26 >  

